h2o-automl/src/main/java/ai/h2o/automl/ModelingStep.java

```diff
@@ -300,6 +300,12 @@ protected void setCommonModelBuilderParams(Model.Parameters params) {
         params._keep_cross_validation_models = buildSpec.build_control.keep_cross_validation_models;
         params._keep_cross_validation_fold_assignment = buildSpec.build_control.nfolds != 0 && buildSpec.build_control.keep_cross_validation_fold_assignment;
         params._export_checkpoints_dir = buildSpec.build_control.export_checkpoints_dir;
+
+        // Give main model main_model_time_budget_factor-times the time to finish than a cv model has; more precisely
+        // main_model_time_budget_factor * cv_max_runtime * nfold/(nfold-1) since the main model uses data from all folds to train.
+        // Main intention is to prevent GLM/DeepLearning non-converging main model to spend
+        // way more time than it should while giving well-behaved models enough time to converge.
+        params._main_model_time_budget_factor = 1;
```

Comment:
I'm thinking about using `2` instead as the multiplier `nfold/(nfold-1)` assumes that the time is linearly dependent on the amount of data (and it is not). But for now `_main_model_time_budget_factor = 1` is good for tests - makes it more likely that something will break.